crypto/ocsp/ocsp_verify.c

```diff
@@ -0,0 +1,380 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <string.h>
+#include "internal.h"
+
+// supporting internal static functions for OCSP_basic_verify
+
+#define IS_OCSP_FLAG_SET(flags, query) (flags & query)
+
+// set up X509_STORE_CTX to verify signer and returns cert chain if verify is OK
+static X509 *ocsp_find_signer_sk(STACK_OF(X509) *certs, OCSP_RESPID *id) {
+  if (certs == NULL || id == NULL) {
+    OPENSSL_PUT_ERROR(OCSP, ERR_R_PASSED_NULL_PARAMETER);
+    return NULL;
+  }
+
+  unsigned char tmphash[SHA_DIGEST_LENGTH], *keyhash;
```

Comment:
nit:  You could move this declaration to later in the function to when you use it.